src/Strategy/Elixir/MixTree.hs

```diff
@@ -0,0 +1,374 @@
+      _ <- takeWhileP (Just "ignored") (not . isEndLine)
+
+      deps <- many $ try $ mixRecurse $ depth + 1
+      pure [MixDep (PackageName dep) (toDependencyVersion =<< version) location (concat deps)]
+
+    mixRecurse :: Int -> Parser [MixDep]
+    mixRecurse depth = do
+      _ <- chunk "\n"
+      mixDep depth
+
+buildGraph :: [MixDep] -> (M.Map PackageName MixDepResolved) -> (M.Map PackageName [DepEnvironment]) -> Graphing Dependency
+buildGraph deps depsResolved depsEnvs = unfold deps subDeps toDependency
+  where
+    toDependency md =
+      Dependency
+        { dependencyType = dType md
+        , dependencyName = dName md depsResolved
+        , dependencyVersion = dVersion md depsResolved
+        , dependencyLocations = []
+        , dependencyEnvironments = fromMaybe [] $ M.lookup (depName md) depsEnvs
+        , dependencyTags = M.empty
+        }
+
+    dType :: MixDep -> DepType
+    dType m = case (depLocation m) of
+      Hex -> HexType
+      Git _ _ -> GitType
+      _ -> UserType
+
+    dName :: MixDep -> M.Map PackageName MixDepResolved -> Text
+    dName m dr = fromMaybe (reqsName) (resolvedGitUrl)
+      where
+        resolvedGitUrl :: Maybe Text
+        resolvedGitUrl = case depResolvedSCM <$> M.lookup (depName m) dr of
+          Just (Git url _) -> Just url
+          _ -> Nothing
+
+        reqsName :: Text
+        reqsName = case depLocation m of
+          Git url _ -> url
+          _ -> unPackageName $ depName m
+
+    -- Favors in order of locked reference, resolved version, and requirement constraint
+    dVersion :: MixDep -> M.Map PackageName MixDepResolved -> Maybe VerConstraint
+    dVersion m dr =
+      asum
+        [ M.lookup (depName m) dr >>= depResolvedVersion
+        , M.lookup (depName m) dr >>= depResolvedRef
+        , rawVersion
+        ]
+      where
+        rawVersion = case (depLocation m) of
+          Git _ (Just ref) -> Just $ CEq ref
+          _ -> depVersion m
+
+-- | Operators used in the mix requirements
+data MixConstraintOperators
+  = GreaterThanOrEqual
+  | GreaterThan
+  | LessThanOrEqual
+  | LessThan
+  | Equal
+  | NotEqual
+  | WildcardAny
+  | Compatible
+
+toDependencyVersion :: Text -> Maybe VerConstraint
+toDependencyVersion dt = case parse parseConstraintExpr "" dt of
+  Left _ -> Nothing
+  Right v -> case v of
+    CEq "*" -> Nothing
+    vc -> Just vc
+
+symbol :: Text -> Parser Text
+symbol = L.symbol sc
+  where
+    sc :: Parser ()
+    sc = L.space (void $ some $ char ' ') empty empty
+
+-- | Consumes whitespace ` ` or tab `\t`
+whitespaceOrTab :: Parser Text
+whitespaceOrTab = takeWhileP (Just "whitespaceOrTab") isSpaceOrTab
+  where
+    isSpaceOrTab c = c == ' ' || c == '\t'
+
+-- | Parses `VerConstraint`
+parseVerConstraint :: Parser VerConstraint
+parseVerConstraint = do
+  operator <- whitespaceOrTab *> parseConstraintOperator <* whitespaceOrTab
+  versionText <- findVersionText <* whitespaceOrTab
+  case operator of
```

Comment:
I've modified the code a bit to collapse multiple function into one. I intent to abstract this piece of code out in future PR, once poetry PR is merged. The parsing for version constraint is fairly similar among them - and I'm presuming it will be for lots of languages. I tried the above code, but it led into test failing for irregular spacing. If you insist I can tackle it in this pr. My preference is that I address this subsequent PR.

Reply:
I was able to make this work. This should have been `Nothing -> return $ CEq versionText` instead of `Nothing -> fail ("Could not recognize mix constraint operator:")`. This is for case when user simply does not use any operator e.g. myPkg: "2.2" vs myPkg: ">2.2"